### DISCUSSION TITLE
Consider moving images to a separate branch?

### DISCUSSION THREAD

**Discussion by IlanCosman** [PatrickF1/fzf.fish#234] [Archive]
Hey Patrick!

I was wondering if you'd be open to the idea of moving readme images from the release branch to a separate branch like Tide does.

The primary advantage here is that download times are faster. Right now, fzf.fish takes ~3s to `fisher install` on my machine, wheras Tide takes about half a second. And since fisher downloads things asynchronously, fzf.fish is likely to always be holding up the horses 😄 It also just seems a bit silly to waste time and bandwidth on useless images.

I know this is an incredibly small and nitpicky thing, but I think improving the user experience, even by a small amount, is worth it.

**Comment by PatrickF1** [comment]
Hi Ilan! Sure I'm down to optimize the install speed of the plugin. Thanks for the idea of using other branches. Is this the conventional way of doing it? I have noticed that some repos are able to utilize some sort of GitHub CDN where if you click on the image it opens up by itself without the normal GitHub UI container around it. Did you also explore that option?

Btw congrats on the new tide releases!

**Comment by IlanCosman** [comment]
> Is this the conventional way of doing it?

I mean, no, not as far as I know, but that's because most repos deliver released products like .exes etc. with anything unneeded stripped out. In contrast, Fisher literally just downloads a zipped folder of the branch. So the best way of doing this would probably be to have a seperate `release` branch, where you just have the things people actually need to install. But since you want people to be able to run `fisher install PatrickF1/fzf.fish`, `main` _is_ the release branch.

> I have noticed that some repos are able to utilize some sort of GitHub CDN where if you click on the image it opens up by itself without the normal GitHub UI container around it. Did you also explore that option?

That's an even less clean solution to the problem. For example, with an assets branch, I can do `../assets/animations/async.gif`, instead of getting the full link. The actual images still have to be in a separate branch/repository, or else you aren't solving the fisher problem.

**Comment by IlanCosman** [comment]
Just as an example, powerlevel10k does this with a separate branch https://github.com/romkatv/powerlevel10k-media. It's especially important there, because he has a ton of images, as well as a few largish font files. I just think having a separate branch is cleaner, because it makes working with the repository more cohesive.

**Comment by PatrickF1** [comment]
Ah I see this technique has been going since at least 2016 (source: https://gist.github.com/joncardasis/e6494afd538a400722545163eb2e1fa5) and is quite popular. I shall do it!

I looked around looking for other techniques and I found this one:
https://github.com/junegunn/i
junegunn/fzf's readme references it like so: https://raw.githubusercontent.com/junegunn/i/master/fzf.png
Longer url and requires keeping around a second repo. I like your technique better.

**Comment by PatrickF1** [comment]
Done in 8d877a973c1fa22f8bedd8b4cf70243ddcd983ac.
